alvarocavalcanti/pierre-decheck
# Docker environment

# What we're fixing

Fixes #61 
Fixes #65 

# What we've done here

- Added Docker and docker-composer support
- Standardized all the tests, making sure all test methods get executed
- Silenced logging messages when executing tests

# How to verify these changes

1. Checkout the branch
1. Type `make help` in the console and verify that all the targets are listed
1. Verify that the commands below work:
    1. `make build`
    1. `make clean`
    1. `make run`
    1. `make up`
    1. `make shell-app`
    1. `make shell-dev`
    1. `make flake8`
    1. `make unittest`
    1. `make test`
1. (If PyCharm Pro is available) Ensure that the docker **sidecar** is executed and then follow the instruction to configure PyCharm to use the remote interpreter. Once it's configured:
    1. Open any Python file, delete an import line, find the warning for missing import, navigate the cursor to it and then hit `Option + Enter`. PyCharm should either offer options to import or add the import right away (if unambiguous)
    1. Navigate the keyboard cursor to a member and then hit `Cmd + B`. You should either be taken to its definition or (if you are already at its definition) a list of its usage will be displayed
    1. Open a test file, navigate the keyboard cursor into a test method and hit `Ctrl + Shift + R`, the selected test should be executed and the results displayed
